source/dofs/dof_renumbering.cc

```diff
@@ -877,39 +877,26 @@ namespace DoFRenumbering
         for (unsigned int c = 0; c < n_buckets; ++c)
           local_dof_count[c] = component_to_dof_map[c].size();
 
-
-        // gather information from all CPUs
-        std::vector<types::global_dof_index> all_dof_counts(
-          fe_collection.n_components() *
-          Utilities::MPI::n_mpi_processes(tria->get_communicator()));
-
-        const int ierr = MPI_Allgather(local_dof_count.data(),
-                                       n_buckets,
-                                       DEAL_II_DOF_INDEX_MPI_TYPE,
-                                       all_dof_counts.data(),
-                                       n_buckets,
-                                       DEAL_II_DOF_INDEX_MPI_TYPE,
-                                       tria->get_communicator());
+        std::vector<types::global_dof_index> prefix_dof_count(n_buckets);
+        const int ierr = MPI_Exscan(local_dof_count.data(),
+                                    prefix_dof_count.data(),
+                                    n_buckets,
+                                    DEAL_II_DOF_INDEX_MPI_TYPE,
+                                    MPI_SUM,
+                                    tria->get_communicator());
         AssertThrowMPI(ierr);
 
-        for (unsigned int i = 0; i < n_buckets; ++i)
-          Assert(
-            all_dof_counts[n_buckets * tria->locally_owned_subdomain() + i] ==
-              local_dof_count[i],
-            ExcInternalError());
+        std::vector<types::global_dof_index> global_dof_count(n_buckets);
+        Utilities::MPI::sum(local_dof_count,
+                            tria->get_communicator(),
+                            global_dof_count);
 
         // calculate shifts
-        unsigned int cumulated = 0;
+        types::global_dof_index cumulated = 0;
```

Comment:
@tjhei It appears that you implicitly spotted a bug in the index shift here! :+1: